internal/commands/vaultsecrets/integrations/create.go

```diff
@@ -0,0 +1,248 @@
+	"github.com/hashicorp/hcp/internal/pkg/profile"
+)
+
+type CreateOpts struct {
+	Ctx     context.Context
+	Profile *profile.Profile
+	IO      iostreams.IOStreams
+	Output  *format.Outputter
+
+	IntegrationName string
+	ConfigFilePath  string
+	PreviewClient   preview_secret_service.ClientService
+	Client          secret_service.ClientService
+}
+
+func NewCmdCreate(ctx *cmd.Context, runF func(*CreateOpts) error) *cmd.Command {
+	opts := &CreateOpts{
+		Ctx:     ctx.ShutdownCtx,
+		Profile: ctx.Profile,
+		IO:      ctx.IO,
+		Output:  ctx.Output,
+
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+		Client:        secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "create",
+		ShortHelp: "Create a new integration.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp vault-secrets integrations create" }} command creates a new Vault Secrets integration.
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Create a new Vault Secrets integration:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets integrations create sample-integration --config-file=path-to-file/config.yaml
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "NAME",
+					Documentation: "The name of the integration to create.",
+				},
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "config-file",
+					DisplayValue: "CONFIG_FILE",
+					Description:  "File path to read integration config data.",
+					Value:        flagvalue.Simple("", &opts.ConfigFilePath),
+					Required:     true,
+				},
+			},
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			opts.IntegrationName = args[0]
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return createRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+type IntegrationConfig struct {
+	Version string
+	Type    IntegrationType
+	Details map[string]string
+}
+
+var (
+	TwilioKeys = []string{"account_sid", "api_key_secret", "api_key_sid"}
+	MongoKeys  = []string{"private_key", "public_key"}
+	AWSKeys    = []string{"audience", "role_arn"}
+	GCPKeys    = []string{"audience", "service_account_email"}
+)
+
+func createRun(opts *CreateOpts) error {
+	// Open the file
+	f, err := os.ReadFile(opts.ConfigFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to open config file: %w", err)
+	}
+
+	var i IntegrationConfig
+	err = yaml.Unmarshal(f, &i)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal config file: %w", err)
+	}
+
+	switch i.Type {
+	case Twilio:
+		missingField := validateDetails(i.Details, TwilioKeys)
+
+		if missingField != "" {
+			return fmt.Errorf("missing required field in the config file: %s", missingField)
+		}
+
+		body := &preview_models.SecretServiceCreateTwilioIntegrationBody{
+			Name:               opts.IntegrationName,
+			TwilioAccountSid:   i.Details["account_sid"],
```

Comment:
thoughts on doing `TwilioKeys[0]` etc. for these fields, instead of `account_sid`? and similarly for the others

Reply:
Hmmm, that's a good point. I'll include 2 other providers in the test cases.